SPECS/systemd/systemd.spec

```diff
@@ -1,10 +1,4 @@
 
-# Temporary until azure-release-common is updated to carry macros.dist that defines %azure
-%if %{defined azure}
-%{error:Once azure macro is defined in macros.dist, please remove this}
-%endif
-%global azure 3
```

Comment:
Does `systemd` still build correctly after this change? I think it doesn't have the `azurelinux-release` package present during the build, so the `azure` macro may not be set.

Reply:
With the changes from https://github.com/microsoft/CBL-Mariner/pull/8013 we will also have these defined by the build tools explicitly (but agreed the buildrequires should also cover it).

Reply:
Adding the BR would cause a circular dependency between `systemd` and `azurelinux-release`. With the changes from #8013, which inject the `azl` and `dist` macros, I think we should be fine with building `systemd` the way it's defined in this commit. We need to work on unifying the source of truth for these macros but that's completely out of scope for this PR and #8013.